pkg/uki/common_test.go

```diff
@@ -0,0 +1,137 @@
+// sign it when needed
+// create the db/dbx efivars on the fly with the proper signatures
+// Use efi.EfivarFs for this
+var _ = Describe("Uki utils", Label("uki", "utils"), func() {
+	var fs v1.FS
+	var logger sdkTypes.KairosLogger
+	var memLog *bytes.Buffer
+	var cleanup func()
+
+	BeforeEach(func() {
+		var err error
+		fs, cleanup, err = vfst.NewTestFS(map[string]interface{}{})
+		Expect(err).Should(BeNil())
+		// create fs with proper setup
+		err = fsutils.MkdirAll(fs, "/sys/firmware/efi/efivars", os.ModeDir|os.ModePerm)
+		file, err := os.ReadFile("tests/fbx64.efi")
+		Expect(err).ToNot(HaveOccurred())
+		err = fs.WriteFile("/efitest.efi", file, os.ModePerm)
+		Expect(err).ToNot(HaveOccurred())
+		file, err = os.ReadFile("tests/fbx64.signed.efi")
+		Expect(err).ToNot(HaveOccurred())
+		err = fs.WriteFile("/efitest.signed.efi", file, os.ModePerm)
+		Expect(err).ToNot(HaveOccurred())
+		memLog = &bytes.Buffer{}
+		logger = sdkTypes.NewBufferLogger(memLog)
+		// Override the Efivars location to point to our fake ones
+		// so the go-uefi lib looks in there
+		fakeEfivars, err := fs.RawPath("/sys/firmware/efi/efivars")
+		Expect(err).ToNot(HaveOccurred())
+		attributes.Efivars = fakeEfivars
+	})
+	AfterEach(func() {
+		cleanup()
+	})
+	It("Fails if it cant find the file to check", func() {
+		err := checkArtifactSignatureIsValid(fs, "/notexists.efi", logger)
+		Expect(err).To(HaveOccurred())
+		Expect(err.Error()).To(ContainSubstring("does not exist"))
+	})
+	It("Fails if the file is not a valid efi file", func() {
+		// Create an empty file
+		err := fs.WriteFile("/nonefi.file", []byte{}, os.ModePerm)
+		Expect(err).ToNot(HaveOccurred())
+		err = checkArtifactSignatureIsValid(fs, "/nonefi.file", logger)
+		Expect(err).To(HaveOccurred())
+		Expect(err.Error()).To(ContainSubstring("not a PE file"))
+	})
+
+	It("Fails if the file to check has no signatures", func() {
+		dbFile := fmt.Sprintf("db-%s", attributes.EFI_IMAGE_SECURITY_DATABASE_GUID.Format())
+		file, err := os.ReadFile("tests/db")
+		Expect(err).ToNot(HaveOccurred())
+		err = fs.WriteFile(filepath.Join("/sys/firmware/efi/efivars", dbFile), file, os.ModePerm)
+		Expect(err).ToNot(HaveOccurred())
+		err = checkArtifactSignatureIsValid(fs, "/efitest.efi", logger)
+		Expect(err).To(HaveOccurred())
+		Expect(err.Error()).To(ContainSubstring("no signatures in the file"))
+	})
+
+	It("Signatures does not match", func() {
+		dbFile := fmt.Sprintf("db-%s", attributes.EFI_IMAGE_SECURITY_DATABASE_GUID.Format())
+		file, err := os.ReadFile("tests/db-wrong")
+		Expect(err).ToNot(HaveOccurred())
+		err = fs.WriteFile(filepath.Join("/sys/firmware/efi/efivars", dbFile), file, os.ModePerm)
+		Expect(err).ToNot(HaveOccurred())
+		err = checkArtifactSignatureIsValid(fs, "/efitest.signed.efi", logger)
+		Expect(err).To(HaveOccurred())
+		Expect(err.Error()).To(ContainSubstring("not ok"))
+	})
+
+	It("Signatures does match in DB", func() {
```

Comment:
I kept reading "does not match" :D 
```suggestion
	It("Signature matches the DB", func() {
```